nix/overlays/isabelle.nix

```diff
@@ -0,0 +1,41 @@
+self: super:
+let
+  pkgs = super;
+
+  pkgsForOldZ3 = import (builtins.fetchTarball {
+    name = "nixos-unstable-z3-old";
+    url = "https://github.com/nixos/nixpkgs/archive/da6eec001ab9437a50d4f91d8c1a3704c193647c.tar.gz";
+    sha256 = "0rzxhi45rxxqw44wf0hvyby993a0gjq2kzygq27rjp11wn71q5kf";
+  }) {};
+  z3old = pkgsForOldZ3.z3;
+
+  pkgsForOldPolyML = import (builtins.fetchTarball {
+    name = "nixos-unstable-polyml-old";
+    url = "https://github.com/nixos/nixpkgs/archive/c381bdc1c542a885bea1ac2209f4dc6449206127.tar.gz";
+    sha256 = "0zwhq69hzhcvnigmvxlkqv8nq825s2m6iwikb7fjsj096kr8q2ds";
+  }) {};
+  polymlPatched = pkgsForOldPolyML.polyml.overrideAttrs(args: pkgs.recurseIntoAttrs(rec {
+    configureFlags = args.configureFlags ++ ["--enable-intinf-as-int"];
+  }));
+
+  pkgsForIsabelle = import (builtins.fetchTarball {
+    name = "nixos-unstable-polyml-old";
+    url = "https://github.com/nixos/nixpkgs/archive/5272327b81ed355bbed5659b8d303cf2979b6953.tar.gz";
+    sha256 = "0182ys095dfx02vl2a20j1hz92dx3mfgz2a6fhn31bqlp1wa8hlq";
+  }) {};
```

Comment:
why not get these three tarballs with niv?

Reply:
Yes. I could do that.